hack/generated/pkg/genruntime/resource_reference.go

```diff
@@ -39,9 +40,10 @@ type ResourceReference struct {
 	// resource to another, and Kubernetes resources are uniquely identified by group, kind, (optionally namespace) and
 	// name - the versions are just giving a different view on the same resource
 
+	// +kubebuilder:validation:Pattern="^/subscriptions/(.+?)/resourcegroups/(.+?)/providers/(.+?)/((.+?)/(.+?))+$"
 	// ARMID is a string of the form /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{resourceProviderNamespace}/{resourceType}/{resourceName}.
 	// ARMID is mutually exclusive with Group, Kind, Namespace and Name.
-	ARMID string `json:"armId"`
+	ARMID string `json:"armId"` // TODO: Note that the pattern for this field is NOT case insensitive... technically we should be but it makes the pattern absolutely terrible [Ss][Uu][Bb][Ss]...
```

Comment:
use case-insensitive flag? `"^(?i:/subscriptions/(.+?)/resourcegroups/(.+?)/providers/(.+?)/((.+?)/(.+?))+)$"`

Reply:
Got this to work -- thanks for the prodding I had tried it before but I think used the wrong format for the specifier